Buildings/Applications/DHC/EnergyTransferStations/Examples/CoolingIndirectClosedBuildingLoop.mo

```diff
@@ -0,0 +1,230 @@
+    annotation (Placement(transformation(extent={{140,70},{160,90}})));
+  Modelica.Blocks.Interfaces.RealOutput dTDis(
+    quantity="TemperatureDifference",
+    unit="degC")
+    "District-side (primary) temperature change"
+    annotation (Placement(transformation(extent={{140,90},{160,110}})));
+  Modelica.Blocks.Sources.Constant TSetCHWS(k=273.15 + 7)
+    "Setpoint temperature for building chilled water supply"
+    annotation (Placement(transformation(extent={{-120,10},{-100,30}})));
+  Buildings.Fluid.Sources.Boundary_pT souDis(
+    redeclare package Medium = Medium,
+    p(displayUnit="Pa") = 300000 + 800,
+    use_T_in=false,
+    T=278.15,
+    nPorts=1)
+    "District (primary) source"
+    annotation (Placement(transformation(extent={{-30,60},{-10,80}})));
+  Buildings.Fluid.Sources.Boundary_pT sinDis(
+    redeclare package Medium = Medium,
+    T=287.15,
+    nPorts=1)
+    "District (primary) sink"
+    annotation (Placement(transformation(extent={{128,60},{108,80}})));
+  Buildings.Applications.DHC.EnergyTransferStations.CoolingIndirect coo(
+    redeclare package Medium = Medium,
+    m1_flow_nominal=mDis_flow_nominal,
+    m2_flow_nominal=mBui_flow_nominal,
+    dp1_nominal(displayUnit="Pa") = 500,
+    dp2_nominal(displayUnit="Pa") = 500,
+    Q_flow_nominal=18514,
+    T_a1_nominal(displayUnit="K") = 278.15,
+    T_a2_nominal(displayUnit="K") = 289.15,
+    dp_nominal(displayUnit="Pa") = 5000,
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    k=0.1,
+    Ti=40,
+    yMax=1,
+    yMin=0,
+    initType=Modelica.Blocks.Types.InitPID.InitialOutput,
+    yCon_start=0,
+    reverseAction=true)
+    "Indirect cooling ETS"
+    annotation (Placement(transformation(extent={{40,10},{60,30}})));
+  Modelica.Blocks.Sources.CombiTimeTable QCoo(
+    table=[0,-10E3; 6,-8E3; 6,-5E3; 12,-2E3; 18,-15E3; 24,-10E3],
+    timeScale=3600,
+    extrapolation=Modelica.Blocks.Types.Extrapolation.Periodic)
+    "Cooling demand"
+    annotation (Placement(transformation(extent={{-120,-30},{-100,-10}})));
+  Buildings.Fluid.HeatExchangers.HeaterCooler_u loa(
+    redeclare package Medium = Medium,
+    allowFlowReversal=false,
+    m_flow_nominal=mBui_flow_nominal,
+    from_dp=false,
+    linearizeFlowResistance=true,
+    show_T=true,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyStateInitial,
+    Q_flow_nominal=-1,
+    dp_nominal=100)
+    "Aggregate building cooling load"
+    annotation (Placement(transformation(extent={{40,-86},{60,-66}})));
+  Modelica.Blocks.Math.Gain gai(k=-1/(cp*(16 - 7)))
+    "Multiplier gain for calculating m_flow"
+    annotation (Placement(transformation(extent={{-60,-20},{-40,0}})));
+  Buildings.Fluid.Movers.FlowControlled_m_flow pumBui(
+    redeclare package Medium = Medium,
+    energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState,
+    allowFlowReversal=false,
+    m_flow_nominal=mBui_flow_nominal,
+    nominalValuesDefineDefaultPressureCurve=true,
+    dp_nominal=0)
```

Comment:
may set this `dp_nominal=6000`? otherwise, there will be a warning:
```
Warning:
========
It is recommended that the volume flow rate versus pressure relation
of the fan or pump satisfies the minimum decrease condition

        (per.pressure.dp[i+1]-per.pressure.dp[i])
d[i] = ------------------------------------------------- < 0
       (per.pressure.V_flow[i+1]-per.pressure.V_flow[i])

 is
  d[1]=0     

Otherwise, a solution to the equations may not exist if the fan or pump speed is reduced.
In this situation, the solver will fail due to non-convergence and
the simulation stops.
... "CoolingIndirectClosedBuildingLoop.mat" creating (simulation result file)
```

Reply:
yes, no problem